### DISCUSSION TITLE
Actual DE ?

### DISCUSSION THREAD

**Discussion by Golem642** [DustinBrett/daedalOS#436] [Ideas]
Random thought but would it be possible to transform this into an actual Desktop Environment?
Like you can install it on a Linux computer and it would work just fine with applications there. I know there would be the need to make connections with the window manager somehow, but once that's figured out could it be possible?

Advantages : web languages are very well known, and html+css specifically could allow for easy theme making for example
Disadvantages : JS = slow compared to compiled languages, but I'm sure there's a way to fix this

**Comment by DustinBrett** [comment]
In time I would like to move in that direction. I am not sure if I would have to rewrite but, but I would be willing to. My main goal would be something that can run on the web and natively or even on bare metal one day.

At the moment it can't work this way, although on one computer in my house I have replaced Windows explorer.exe with Edge running my site in full screen, so it feels like the computer boots into daedalOS.

There are things like [greenfield](https://github.com/udevbe/greenfield) which is basically a Wayland compositor and can do what you are asking, but I have not integrated this into my app yet.

I still have a lot of things I haven't done with my project that I hope to get to one day, but at least for the last few months I have slowed down with it. It's probably an endless project so hopefully within my lifetime I can realize all the hopes and dreams I have for it.

Thanks for asking!
